modules/documents/app/models/document_workflow.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class DocumentWorkflow < ApplicationRecord
+  belongs_to :type, class_name: "DocumentType"
+  belongs_to :old_status, class_name: "DocumentStatus"
+  belongs_to :new_status, class_name: "DocumentStatus"
+  belongs_to :role
+end
```

Comment:
I guess asking here for the sake of completeness, but we talked about using Workflow + Status + Type and jus creating document specialisation for those models. Here we created them isolated.

Anything specific that skewed the decision? This looks absolutely fine to me, asking out of curiosity, because it felt like we could benefit from the original models.